### PR TITLE
[Serializer] Fix unknown expected types normalization the Promotion constructor type when know type

### PR DESCRIPTION
**Pull request by Myks92**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Issues        | -
| License       | MIT

When we use the normalizer with Promotion Constructor, a Non Normalizable Value Exception occurs with unknown type even when the type is known. This Pull request fix this error.

### CODE REVIEW

**Review by nicolas-grekas — APPROVED** [review]
LGTM, for branch 5.4

File: src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php [hunk 1]
```diff
@@ -377,10 +377,16 @@ protected function instantiateObject(array &$data, string $class, array &$contex
                         continue;
                     }
 
+                    $constructorParameterType = 'unknown';
+                    $reflectionType = $constructorParameter->getType();
+                    if (null !== $reflectionType && method_exists($reflectionType, 'getName')) {
```
Comment (mtarld) [inline src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php]:
```suggestion
                    if ($reflectionType instanceof \ReflectionNamedType) {
```

**Review by mtarld — CHANGES_REQUESTED** [review]

**Review by mtarld — APPROVED** [review]
LGTM, but you should target the 5.4 branch as that bug is already existing in that version

**Review by lyrixx — APPROVED** [review]
Awesome 👍🏼

**Pull request closed without merging**